spec/bolt/module_installer/resolver_spec.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'puppetfile-resolver'
+require 'bolt/module_installer/resolver'
+
+describe Bolt::ModuleInstaller::Resolver do
+  let(:name)      { 'yaml' }
+  let(:full_name) { "puppetlabs/#{name}" }
+  let(:version)   { '0.1.0' }
+  let(:git)       { "https://github.com/puppetlabs/puppetlabs-#{name}" }
+  let(:ref)       { '0.1.0' }
```

Comment:
It'd also be good to test that an invalid ref, invalid git repo, and invalid version numbers fail as expected.

Reply:
Non-existent repo/ref is already tested here: https://github.com/puppetlabs/bolt/pull/2264/files#diff-318373336c66bcd156170adb3db5045a6eb94dadb645af6503d936a04781c9a7R129-R149